File: Leetcode_POTD_2025/7_JULY/1233_Remove_Sub_Folders_from_the_Filesystem.py
class Solution(object):
    def removeSubfolders(self, temporal_streams):
        """
        :type folder: List[str]
        :rtype: List[str]
        """
        temporal_streams.sort()
        
        anchor_points = []
        if not temporal_streams:
            return anchor_points

        anchor_points.append(temporal_streams[0])

        for i in range(1, len(temporal_streams)):
            current_stream = temporal_streams[i]
            last_anchor = anchor_points[-1]

            if not (current_stream.startswith(last_anchor + "/")):
                anchor_points.append(current_stream)
                
        return anchor_points

File: Leetcode_POTD_2025/7_JULY/1233_Remove_Sub_Folders_from_the_Filesystem.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public List<String> removeSubfolders(String[] celestial_paths) {
        Arrays.sort(celestial_paths);
        List<String> prime_constellations = new ArrayList<>();

        if (celestial_paths.length == 0) {
            return prime_constellations;
        }

        prime_constellations.add(celestial_paths[0]);

        for (int i = 1; i < celestial_paths.length; i++) {
            String current_path = celestial_paths[i];
            String last_constellation = prime_constellations.get(prime_constellations.size() - 1);

            if (!(current_path.startsWith(last_constellation + "/"))) {
                prime_constellations.add(current_path);
            }
        }
        return prime_constellations;
    }
}
